supabase/migrations/43_payment_emails.sql

```diff
@@ -0,0 +1,292 @@
+begin;
+
+create or replace type alert_type as enum (
+  'free_trial',
+  'free_trial_ending',
+  'free_trial_stalled',
+  'missing_payment_method',
+  'data_movement_stalled',
+  'data_not_processed_in_interval' -- Old alert type
+);
+
+-- In order to allow alerts to contain arguments after they're done firing
+-- we need to refactor alerts to contain a `firing` boolean, rather than
+-- simply omitting no-longer-firing alerts from the view.
+create type alert_snapshot as (
+  alert_type alert_type,
+  catalog_name catalog_name,
+  arguments json,
+  firing boolean
+);
+
+create or replace view internal.alert_free_trial as
+select
+  'free_trial'::alert_type as alert_type,
+  (tenants.tenant || 'alerts/free_trial')::catalog_name as catalog_name,
+  json_build_object(
+    'tenant', tenants.tenant,
+    'recipients', array_agg(json_build_object(
+      'email', alert_subscriptions.email,
+      'full_name', auth.users.raw_user_meta_data->>'full_name'
+    )),
+    'trial_start', tenants.trial_start::date,
+    'trial_end', (tenants.trial_start + interval '1 month')::date,
+    'has_credit_card', stripe.customers."invoice_settings/default_payment_method" is not null
+  ) as arguments,
+  -- Since we don't need to communicate post-alert arguments, we can instead
+  -- simply omit tenants that are no longer in their free trial, and mark the
+  -- those that are as firing.
+  true as firing
+from tenants
+  left join alert_subscriptions on alert_subscriptions.catalog_prefix ^@ tenants.tenant and email is not null
+  left join stripe.customers on stripe.customers."name" = tenants.tenant
```

Comment:
no action, i think: just noting we'll need to account for google cloud marketplace here in the future.

More generally, there can be payment methods other than stripe.